### PR TITLE
Replace deprecated HasTraits.set with trait_set

### PR DESCRIPTION
**Pull request by effigies**
Tests are currently failing, I think at least partially because of a bunch of deprecation warnings about `HasTraits.set`.

This fixes that (I think). Then hopefully the logs will be clear enough to identify other issues.

### CODE REVIEW

**Comment by effigies** [comment]
Cleaned up some simple nibabel-related warnings.

File: circle.yml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 machine:
+  pre:
+    - curl -sSL https://s3.amazonaws.com/circle-downloads/install-circleci-docker.sh | bash -s -- 1.10.0
```
Comment (oesteban) [inline circle.yml]: This would break the cache (if it worked with 1.9, we didn't get it to work, right @satra?). Caching does not work with 1.10.

Comment (satra) [inline circle.yml]: sth weird happened with docker. i thought i had removed 1.9 and caching worked, but not after it got merged. we still have to revisit the cache situation. from a time perspective restoring and saving the cache takes almost as time as building it.

Comment (oesteban) [inline circle.yml]: yes, we should probably roll back to the docker pull solution, and manually update nipype/base

Comment (oesteban) [inline circle.yml]: (I don't have hopes for get the cache to work. Definitely not with 1.10 but 1.9 neither)

**Pull request merged**